test/models/namespace_bot_test.rb

```diff
@@ -33,4 +33,30 @@ def setup
       namespace_bot.destroy
     end
   end
+
+  test 'validate_bot_user_type' do
+    valid_namespace_bot = namespace_bots(:project1_bot0)
+    assert_not valid_namespace_bot.errors.full_messages.any?
+
+    valid_namespace_bot.validate_bot_user_type
+    assert_not valid_namespace_bot.errors.full_messages.any?
+
+    invalid_user_bot = namespace_bots(:invalid_user_bot)
+    assert_not invalid_user_bot.errors.full_messages.any?
+
+    invalid_user_bot.validate_bot_user_type
+    assert invalid_user_bot.errors.full_messages.any?
+  end
+
+  test 'remove_membership_from_namespace' do
+    namespace_bot = namespace_bots(:project1_bot0)
+    membership = namespace_bot.user.members.find_by(namespace: namespace_bot.namespace)
+
+    assert_not_nil membership
+    assert namespace_bot.membership
+    assert_equal membership, namespace_bot.membership
+
+    namespace_bot.remove_membership_from_namespace
```

Comment:
Currently the method `remove_membership_from_namespace` in `namespace_bot.rb` isn't working correctly with the `membership&.destroy`. If you update that to `user.members.find_by(namespace:).destroy` the method works and the underlying membership is removed

Reply:
Thanks, fixed in 2ff15e6a8e95b7f8df2181cf248fc0bbd039d04b